pkg/registry/core/componentstatus/rest.go

```diff
@@ -70,13 +74,47 @@ func (rs *REST) List(ctx context.Context, options *metainternalversion.ListOptio
 	wait.Wait()
 	close(statuses)
 
+	pred := componentStatusPredicate(options)
```

Comment:
we can start by supporting only label-selecting, and this will help simplify this pull, wdyt?

Reply:
Hi @yue9944882,
thx for your review. IMO the changes are quite simple as it is, is some 30 lines of code which is fairly straightforward. More importantly, taking out the field selectors is hardly going to change the code and will not satisfy the requirement from the ticket.